src/seedu/addressbook/data/person/Block.java

```diff
@@ -0,0 +1,51 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's block in a Person's Address the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidBlockAddress(String)}
+ */
+public class Block {
+    public static final String EXAMPLE = "123";
+    public static final String MESSAGE_ADDRESS_BLOCK_CONSTRAINTS = "Address Block can be in any format";
+    public static final String ADDRESS_BLOCK_VALIDATION_REGEX = ".+";
+
+    public final String value;
+
+    /**
+     * Validates given block address.
+     *
+     * @throws IllegalValueException if given block address string is invalid.
+     */
+    public Block(String block) throws IllegalValueException {
+        if (!isValidBlockAddress(block)) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_BLOCK_CONSTRAINTS);
+        }
+        this.value = block;
+    }
+
+    /**
+     * Returns true if a given string is a valid block address
+     */
+    public static boolean isValidBlockAddress(String test) {
+        return test.matches(ADDRESS_BLOCK_VALIDATION_REGEX);
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Why do you need to add `@Override` here and other similar methods, e.g. `toString`, `equals`? Think why and why not?